src/mono/mono.proj

```diff
@@ -31,6 +31,9 @@
     <BuildMonoAOTCrossCompiler Condition="'$(TargetsBrowser)' == 'true'">true</BuildMonoAOTCrossCompiler>
     <CrossConfigH Condition="'$(BuildMonoAOTCrossCompiler)' == 'true'">$(MonoObjDir)cross\config.h</CrossConfigH>
     <MonoBundleLLVMOptimizer Condition="'$(MonoEnableLLVM)' == 'true'">true</MonoBundleLLVMOptimizer>
+    <MonoCCompiler>$(Compiler)</MonoCCompiler>
+    <MonoCCompiler Condition="'$(MonoCCompiler)' == ''">clang</MonoCCompiler>
+    <MonoCCompiler Condition="$(MonoCCompiler.StartsWith('-'))">$(MonoCCompiler.TrimStart('-'))</MonoCCompiler>
```

Comment:
When we use `runtime/build.sh -gcc`, it is passed to `eng/build.sh` which sets the raw value to `$(Compiler)`.

In case of coreclr, libraries and installer, these values are eventually passed to `build-commons.sh` (build.sh -> proj -> {native-build-script-name}.sh -> build-commons.sh) and get stripped by the time `init-compiler.sh` is called. Here we have a flat hierarchy (i.e. easy to analyze the transformation to expected value).

Reply:
Ok, makes sense. Please add that as a comment in the source code.

Reply:
Added a comment above.